ledger/src/blockstore.rs

```diff
@@ -301,54 +301,65 @@ impl Blockstore {
         false
     }
 
-    /// Silently deletes all blockstore column families starting at the given slot until the `to` slot
+    /// Silently deletes all blockstore column families in the range [from_slot,to_slot]
     /// Dangerous; Use with care:
     /// Does not check for integrity and does not update slot metas that refer to deleted slots
     /// Modifies multiple column families simultaneously
-    pub fn purge_slots(&self, mut from_slot: Slot, to_slot: Option<Slot>) {
+    pub fn purge_slots_with_delay(
+        &self,
+        from_slot: Slot,
+        to_slot: Slot,
+        delay_between_purges: Option<Duration>,
+    ) {
         // if there's no upper bound, split the purge request into batches of 1000 slots
         const PURGE_BATCH_SIZE: u64 = 1000;
-        let mut batch_end = to_slot.unwrap_or(from_slot + PURGE_BATCH_SIZE);
-        while from_slot < batch_end {
-            match self.run_purge(from_slot, batch_end) {
-                Ok(end) => {
-                    if !self.no_compaction {
-                        if let Err(e) = self.compact_storage(from_slot, batch_end) {
-                            // This error is not fatal and indicates an internal error
-                            error!(
-                                "Error: {:?}; Couldn't compact storage from {:?} to {:?}",
-                                e, from_slot, batch_end
-                            );
-                        }
-                    }
-
-                    if end {
-                        break;
-                    } else {
```

Comment:
This appears to be completely broken.  "end" is really "all_columns_purge", and regardless of true/false we don't want to `break` here